src/python/gudhi/_edge_collapse.cc

```diff
@@ -0,0 +1,66 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Marc Glisse
+ *
+ *    Copyright (C) 2023 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <vector>
+#include <utility>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+
+#include <gudhi/Flag_complex_edge_collapser.h>
+
+namespace py = pybind11;
+
+template<class Index, class Filtr>
+py::object collapse(py::array_t<Index> is, py::array_t<Index> js, py::array_t<Filtr> fs, int nb_iterations) {
+  typedef std::tuple<Index, Index, Filtr> Filtered_edge;
+  typedef std::vector<Filtered_edge> Edges;
+  py::buffer_info bufi = is.request();
+  py::buffer_info bufj = js.request();
+  py::buffer_info buff = fs.request();
+  if (bufi.ndim != 1 || bufj.ndim != 1 || buff.ndim != 1)
+    throw std::runtime_error("Input must be 1-dimensional arrays");
+  if (bufi.shape[0] != bufj.shape[0] || bufi.shape[0] != buff.shape[0])
+    throw std::runtime_error("Input arrays must have the same size");
+  if (buff.shape[0] == 0) {
+    py::array_t<Index> indices({{ 2, 0 }}, {{ 0, 0 }});
+    py::array_t<Filtr> filtrs;
+    return py::make_tuple(std::move(indices), std::move(filtrs));
+  }
+  auto& edges = *new Edges();
+  {
+    py::gil_scoped_release release;
+    edges.reserve(bufi.shape[0]);
+    Index n_edges = static_cast<Index>(bufi.shape[0]);
```

Comment:
I don't think it costs to call `bufi.shape[0]`, but it could be:
```suggestion
    Index n_edges = static_cast<Index>(bufi.shape[0]);
    edges.reserve(n_edges);
```